hocs/withNoAuthRoute.tsx

```diff
@@ -0,0 +1,30 @@
+import { FunctionComponent, useEffect } from 'react';
+import { useRouter } from 'next/router';
+import { useComponentDidMount, useRecoilValueAfterMount } from 'hooks';
+import userState from 'core';
+
+/**
+ * NOTE: 해당 HOC는 auth/signin, auth/signup 등
+ * 로그인 하지 않은 페이지에서만 사용할 것
+ */
+function withNoAuth<P>(Component: FunctionComponent<P>) {
+  return function WithAuthComponent(props: P) {
+    const mounted = useComponentDidMount();
+    const user = useRecoilValueAfterMount(userState, null);
+    const router = useRouter();
+
+    useEffect(() => {
+      if (!mounted) return;
+      if (user && !user.isCouple) {
+        router.push('/profile');
+        return;
+      }
+      router.push('/');
```

Comment:
```suggestion
      if (user) {
        if (user.isCouple) router.replace('/');
        else router.replace('/profile');
      }
```

일단 준혁님 코드와 비슷한데 저는 이렇게 처리했는데, 

유저가 있는지 확인하고 그 안에서 커플인지 아닌지를 판단하는게 더 읽기 쉬울것 같아 위와 같이 작성했는데 어떠신가요?